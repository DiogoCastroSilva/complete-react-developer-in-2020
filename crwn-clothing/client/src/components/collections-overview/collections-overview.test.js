import React from 'react';
import { shallow } from 'enzyme';

import CollectionsOverview from './collections-overview.component';
import CollectionPreview from '../collection-preview/collection-preview.component';


describe('testing CollectionsOverview component', () => {
    let wrapper;

    beforeEach(() => {
        const mockProps = {
            collections: [
                {
                    id: 1,
                    title: 'hats',
                    items: []
                }
            ]
        };

        wrapper = shallow(<CollectionsOverview {...mockProps} />);
    });

    it('should render a CollectionPreview component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have one CollectionPreview component', () => {
        expect(wrapper.find(CollectionPreview).length).toEqual(1);
    });

    it('should not have any CollectionPreview component', () => {
        const withItemsMockProps = {
            collections: []
        };

        const newWrapper = shallow(<CollectionsOverview {...withItemsMockProps} />);

        expect(newWrapper.find(CollectionPreview).exists()).toEqual(false);
    });
});