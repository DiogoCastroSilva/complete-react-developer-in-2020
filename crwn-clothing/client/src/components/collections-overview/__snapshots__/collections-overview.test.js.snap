// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CollectionsOverview component should render a CollectionPreview component 1`] = `
<styled.div>
  <withRouter(CollectionPreview)
    items={Array []}
    key="1"
    title="hats"
  />
</styled.div>
`;
